src/components/Map/BasemapLayer.js

```diff
@@ -0,0 +1,42 @@
+import React, {
+  useEffect, useCallback, useRef
+} from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import { vectorBasemapLayer } from 'esri-leaflet-vector';
+import { mapConfig } from '../../configuration/config';
+
+const basemaps = mapConfig.basemaps;
+
+const baseMapSelector = (state) => state.mapProperties.basemap;
+
+export default function BasemapLayer(props) {
+  const { map } = props;
+  const selectedBasemap = useSelector(baseMapSelector);
+  const basemapRef = useRef(null);
+  const handleBasemapChange = useCallback((basemapName) => {
+    if (basemapRef.current !== null) {
+      basemapRef.current.remove(map);
+    }
+    if (map) {
+      const newBasemap = vectorBasemapLayer(basemaps[basemapName], {
+        apikey: 'AAPK19aa44a23e4c4a7788b37541444c07denQmRCIZSmoomcyzoK3i3ko37EDOjKQcz9ui0gt3KdWAk6c2N4fPW9jEBF0__3R7o',
```

Comment:
We'll want to make this a github secret and probably also get a new API key since this is in our commit history now

Reply:
Actually, you can limit this key in AGOL to a specific domain, so there's no need to hide it. I already verified that it works.

Reply:
Ah okay I don't think I knew that. Nevermind then 😄